### PR TITLE
Update website build to nbsite

### PR DESCRIPTION
**Pull request by philippjfr**
Makes use of ``nbsite.gallery`` (https://github.com/pyviz/nbsite/pull/106)

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -3,42 +3,65 @@
...
+pyviz_module = 'holoviews'
 description = 'Stop plotting your data - annotate your data and let it visualize itself.'
 
 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
 # built documents.
 #
 # The short X.Y version.
-version = setup_args['version']
+version = 'v1.11.0' #setup_args['version']
```
Comment (jsignell) [inline doc/conf.py]: Is the version not accessible?

File: doc/conf.py [hunk 2]
```diff
@@ -3,42 +3,65 @@
...
+                'elements',
+                'containers',
+                'streams',
+                'apps'
+            ],
+            'title': 'Reference Gallery',
+        }
+    },
+    'github_org': 'pyviz',
+    'github_project': 'holoviews'
```
Comment (jsignell) [inline doc/conf.py]: could these be a level higher and be used by generate-rst and build as well?

Comment (philippjfr) [reply doc/conf.py]: Probably.

File: doc/conf.py (hunk 1 above)
Comment (philippjfr) [reply doc/conf.py]: I had trouble with it locally but will try to restore it before merging.

File: doc/conf.py (hunk 2 above)
Comment (jsignell) [reply doc/conf.py]: Ok that would be a good goal for nbsite. Then we could just write a `doit build_docs` in pyctdev and wouldn't need to run `nbsite generate-rst` and `nbsite build` with all the flags everytime.

File: doc/conf.py (hunk 1 above)
Comment (jsignell) [reply doc/conf.py]: This is what we do on our autover projects 

```python
import colorcet
version = release = colorcet.__version__
```

File: doc/conf.py (hunk 2 above)
Comment (philippjfr) [reply doc/conf.py]: Should handle this in nbsite and then update it here. Will merge this as is for now.

**Comment by jlstevens** [comment]
Looks good and I am happy to see this merged. It will be good once holoviews.org can be built entirely with nbsite (with a nice looking twitter feed that doesn't need manual tweaking!)

**Pull request merged**